contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/afterSimAnalysis/StayTaskManager.java

```diff
@@ -0,0 +1,109 @@
+package org.matsim.contrib.drt.analysis.afterSimAnalysis;
+
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.events.ActivityEndEvent;
+import org.matsim.api.core.v01.events.ActivityStartEvent;
+import org.matsim.api.core.v01.events.handler.ActivityEndEventHandler;
+import org.matsim.api.core.v01.events.handler.ActivityStartEventHandler;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.population.Person;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class StayTaskManager implements ActivityStartEventHandler, ActivityEndEventHandler {
+
+	private final Map<Id<Person>, StayTaskDataEntry> startedSatyTasksMap = new HashMap<>();
+	private final List<StayTaskDataEntry> stayTaskDataEntriesList = new ArrayList<>();
+	private int counter = 0;
+
+	class StayTaskDataEntry {
+		private Id<Link> linkId;
+		private double StartTime;
+		private double EndTime;
```

Comment:
`startTime` and `endTime`?

Reply:
`startTime` and `linkId` could be final

Reply:
The startTime and endTime refer to the starting time and ending time of the stay task.

Reply:
Sorry, that wasn't clear enough. I meant changing `StartTime` and `EndTime` to `startTime` and `endTime`.